$MirOS: ports/misc/mc/patches/patch-vfs_extfs_mailfs_in,v 1.1 2008/09/17 20:06:59 tg Exp $
--- vfs/extfs/mailfs.in.orig	Tue Dec 24 06:55:55 2002
+++ vfs/extfs/mailfs.in	Fri Dec 31 17:44:06 2010
@@ -7,6 +7,8 @@ use bytes;
 
 $zcat="zcat";                 # gunzip to stdout
 $bzcat="bzip2 -dc";           # bunzip2 to stdout
+$lzcat="lzma -dc";            # unlzma to stdout
+$xzcat="xz -dc";              # unxz to stdout
 $file="file";                 # "file" command
 $TZ='GMT';                    # default timezone (for Date module)
 
@@ -132,6 +134,10 @@ if (/gzip/) {
     exit 1 unless (open IN, "$zcat $mbox_qname|");
 } elsif (/bzip/) {
     exit 1 unless (open IN, "$bzcat $mbox_qname|");
+} elsif (/xz/) {
+    exit 1 unless (open IN, "$xzcat $mbox_qname|");
+} elsif (/lzma/) {
+    exit 1 unless (open IN, "$lzcat $mbox_qname|");
 } else {
     exit 1 unless (open IN, "<$mbox_name");
 }
